cmd/juju/application/deployer/deployer.go

```diff
@@ -272,24 +278,27 @@ func (d *factory) localPreDeployedCharmDeployer() (DeployerKind, error) {
 	if !charm.Local.Matches(userCharmURL.Schema) {
 		return nil, errors.Errorf("cannot interpret as a redeployment of a local charm from the controller")
 	}
-	return &localPreDeployerKind{userCharmURL: userCharmURL}, nil
+
+	charmInfo, err := deployAPI.CharmInfo(userCharmURL.String())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	ch := charmInfo.Charm()
+
+	// Determine base
+	base, _, err := d.determineBaseForCharm(ch, deployAPI)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
```

Comment:
I'm assuming the series in the charm URL made this deterministic. If it wasn't present we'd fallback to something?

Your changes explicitly call out that the `localPreDeployerKind` and `predeployedLocalCharm` needs a base now.

I'm surprised if we're not already making this call for the charm URL (identifier elsewhere) and we could cache it. Something to think about in the future.

Reply:
If a series wasn't present in a charm url we would fail